SnapcallListener.js

```diff
@@ -0,0 +1,183 @@
+  console.log("onUnheld", parameter);
+  callEvent("onUnheld", parameter)
+}
+
+function onTime(parameter) {
+  console.log("onTime", parameter, typeof parameter);
+
+  callEvent("onTime", parameter)
+}
+
+function onConnectionShutDown() {
+  console.log("onConnectionShutDown");
+  var i = 0;
+  while  (i < eventListener["onConnectionShutDown"].length){
+    eventListener["onConnectionShutDown"][i]();
+    i++;
+  }
+}
+
+function callEvent(eventName, parameter) {
+  var eventObject = null;
+  console.log(parameter);
+  if (typeof parameter.data == "string") {
+    console.log("detec")
+     eventObject = JSON.parse(parameter.data) ;
+  }
+  else if (parameter) {
+    console.log("detec 2")
+    eventObject = parameter.data
+  }
+  console.log("event paramter :",  eventObject);
+  var i = 0;
+  while  (i < eventListener[eventName].length){
+    eventListener[eventName][i](eventObject);
+    i++;
+  }
+}
+
+const eventListener = {
+
+  "onConnectionReady": [],
+  "onCreated":  [],
+  "onRinging": [],
+  "onAnswer": [],
+  "onInternetDown": [],
+  "onInternetUP": [],
+  "onHangup": [],
+  "onMuteChange": [],
+  "onSpeakerChange": [],
+  "onHeld": [],
+  "onUnheld": [],
+  "onTime": [],
+  "onConnectionShutDown": []
+};
+
+function getData(parameter) {
+  if (parameter) {
+    const clientEvent = Json.parse(parameter);
+    return clientEvent.data;
+  }
+  return null;
+}
+function callEventListener(eventName, parameter) {
+
+  const data = getData(parameter);
+  var i = 0;
+
+  if (eventListener[eventName]) {
+    while (i < eventListener[eventName]) {
+      eventListener[eventName][i](data);
+      i++;
+    }
+  }
+}
+
+export class SnapcallListener {
+
+  subscription = []
+
+  constructor(isIOS) {
+    eventEmitter = isIOS ? new NativeEventEmitter(NativeModules.RNSnapcallEmitEvent) : DeviceEventEmitter;
+
+    this.subscription.push(eventEmitter.addListener("onConnectionReady", onConnectionReady));
+    this.subscription.push(eventEmitter.addListener("onCreated", onCreated));
+    this.subscription.push(eventEmitter.addListener("onRinging", onRinging));
+    this.subscription.push(eventEmitter.addListener("onAnswer", onAnswer));
+    this.subscription.push(eventEmitter.addListener("onInternetDown", onInternetDown));
+    this.subscription.push(eventEmitter.addListener("onInternetUP", onInternetUP));
+    this.subscription.push(eventEmitter.addListener("onHangup", onHangup));
+    this.subscription.push(eventEmitter.addListener("onHeld", onHeld));
+    this.subscription.push(eventEmitter.addListener("onUnheld", onUnheld));
+    this.subscription.push(eventEmitter.addListener("onMuteChange", onMuteChange));
+    this.subscription.push(eventEmitter.addListener("onSpeakerChange", onSpeakerChange));
+    this.subscription.push(eventEmitter.addListener("onTime", onTime));
+    this.subscription.push(eventEmitter.addListener("onConnectionShutDown", onConnectionShutDown));
+  }
+
+  release() {
+    while (this.subscription[0]) {
```

Comment:
I'm wondering why you need a `while` here, are you planning to delete multiple subscriptions or just the first one?
If it's the first one, you could do something like this?
```if (this.subscription[0]) {
  this.subscription[0].remove();'
  delete this.subscription[0];
}```

Reply:
i remove all the element here